job/server/src/main/java/alluxio/master/job/plan/PlanCoordinator.java

```diff
@@ -123,9 +124,9 @@ private synchronized void start() throws JobDoesNotExistException {
     } catch (Exception e) {
       LOG.warn("Failed to select executor. {})", e.toString());
       LOG.info("Exception: ", e);
-      mPlanInfo.setStatus(Status.FAILED);
       mPlanInfo.setErrorType(ErrorUtils.getErrorType(e));
       mPlanInfo.setErrorMessage(e.getMessage());
+      mPlanInfo.setStatus(Status.FAILED);
```

Comment:
What is the reason you put `mPlanInfo.setStatus(Status.FAILED);`  to the last .

Reply:
yeah, we should really use the convenience method to set status to failed so we don't forget to set error message and types.  fixed